pkg/apis/v1alpha1/cluster_supply_chain.go

```diff
@@ -254,27 +144,58 @@ type SupplyChainResource struct {
 	Configs []ResourceReference `json:"configs,omitempty"`
 }
 
-var ValidSupplyChainTemplates = []client.Object{
-	&ClusterSourceTemplate{},
-	&ClusterImageTemplate{},
-	&ClusterConfigTemplate{},
-	&ClusterTemplate{},
-}
-
 type SupplyChainTemplateReference struct {
 	// Kind of the template to apply
 	//+kubebuilder:validation:Enum=ClusterSourceTemplate;ClusterImageTemplate;ClusterTemplate;ClusterConfigTemplate
 	Kind string `json:"kind"`
+
+	// Name of the template to apply
+	// Only one of Name and Options can be specified.
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
```

Comment:
I noticed that here we're specifying the `MinLength=1` constraint, but in other places like SupplyChainResource we don't

```go
  type SupplyChainResource struct {
          // Name of the resource. Used as a reference for inputs, as well as being
          // the name presented in workload statuses to identify this resource.
          Name string `json:"name"`
```

do you think we should start making use of this more broadly? e.g,. currently it's possible to have a SupplyChainResource with `""` (empty string) set

```yaml
  apiVersion: carto.run/v1alpha1
  kind: ClusterSupplyChain
  metadata:
    name: sc
  spec:
    selector:
      foo: bar
    resources:
      - name: ""
        templateRef:
          kind: ClusterSourceTemplate
          name: foo
```

(might be worth a separate issue?)